### ISSUE TITLE
tf.keras.preprocessing.image_dataset_from_directory ignores labels passed in 'labels' argument

### ISSUE THREAD

**Issue by Justus-M** [tensorflow/tensorflow#44770]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
binary
- TensorFlow version (use command below):
tensorflow-gpu 2.3.0
- Python version:
3.8.3
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
CUDA Version: 10.1
- GPU model and memory:
4 x Tesla P100 16GB

**Describe the current behavior**
tf.keras.preprocessing.image_dataset_from_directory creates labels based on the subdirectories despite labels being passed explicitly. Despite labels from 6 different classes being passed, only 3 are recognized (from the folder structure). The issue persists if I only have one subfolder in the directory.
**Describe the expected behavior**
The expectation is for the labels not to be ignored, and to be used instead of trying to infer them from the folder structure.
**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
train = tf.keras.preprocessing.image_dataset_from_directory(
                    '/home/jupyter/JM/images', labels=list(np.uint8(np.round(np.random.random(6613)*5))),
    image_size=(1024, 1024), validation_split=0.3, subset = 'training', seed = 42, label_mode = 'categorical')

file path is irrelevant - just choose a viable file path with subdirectories and replace the 6613 with the number of files in the subdirectories

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
Found 6613 files belonging to 3 classes.

**Comment by bhack** [comment]
I've started to fix https://github.com/tensorflow/tensorflow/issues/44752 with https://github.com/tensorflow/tensorflow/pull/44769.

You problem is  `num_classes=len(class_names)` I will try to fix after i got a feedback at https://github.com/tensorflow/tensorflow/pull/44769.

**Comment by justinmulli** [comment]
@ravikyram is this being resolved?

**Comment by Saduf2019** [comment]
@Justus-M 
Could you please check with the latest tf version and update us, as the PR is merged this should be resolved,
